rust/template/differential_datalog/src/dataflow/filter_map.rs

```diff
@@ -4,19 +4,23 @@ use timely::{
     Data,
 };
 
+pub type FilterMapFunc<D, D2> = std::sync::Arc<dyn Fn(D) -> Option<D2>>;
+
 pub trait FilterMap<D, D2> {
     type Output;
 
-    fn filter_map<L>(&self, logic: L) -> Self::Output
```

Comment:
You don't need to alter the dataflow operators to do this

Reply:
I'm a little confused with this comment. Which part I don't need to alter here?

Reply:
The function signatures here shouldn't change, I'll make code suggestions